### PR TITLE
Add API to control logs for animators

### PR DESCRIPTION
**Pull request by kiryldz**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).
Fixes: #243 

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [x] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Add example if relevant.
 - [x] Document any changes to public APIs.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [x] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog>Add API to control logging for animation plugin and disable debug logs by default.</changelog>`.

### Summary of changes

Add API to disable not critical logs for camera plugin. `debugMode` is set to false by default. Logs could be enabled with:

```
mapView.camera.debugMode = true
```

Additionally even if logs are enabled non-critical logs are downgraded to use `Logger.d` instead of `Logger.i`.

### CODE REVIEW

File: plugin-animation/src/main/java/com/mapbox/maps/plugin/animation/CameraAnimationsPluginImpl.kt [hunk 1]
```diff
@@ -57,6 +57,8 @@ internal class CameraAnimationsPluginImpl : CameraAnimationsPlugin {
 
   private val lifecycleListeners = CopyOnWriteArraySet<CameraAnimationsLifecycleListener>()
 
+  override var debugMode: Boolean = true
```
Comment (Chaoba) [inline plugin-animation/src/main/java/com/mapbox/maps/plugin/animation/CameraAnimationsPluginImpl.kt:60]: Can we make it false by default? Users who want to see the log can enable it intentionally, but I don't think it's a common requirement for most users.

File: plugin-animation/src/test/java/com/mapbox/maps/plugin/animation/CameraAnimationsPluginImplTest.kt [hunk 2]
```diff
@@ -962,6 +963,21 @@ class CameraAnimationsPluginImplTest {
     assertEquals(10.0, updateList[1].pitch!!, EPS)
   }
 
+  @Test
+  fun debugModeTest() {
+    // debug mode is enabled by default for backward compatibility
+    shadowOf(getMainLooper()).pause()
+    cameraAnimationsPluginImpl.easeTo(cameraOptions, mapAnimationOptions { duration(DURATION) })
+    shadowOf(getMainLooper()).idle()
+    assertNotEquals(0, ShadowLogger.logCount)
```
Comment (Chaoba) [inline plugin-animation/src/test/java/com/mapbox/maps/plugin/animation/CameraAnimationsPluginImplTest.kt:972]: If I understand correctly when debug mode is enabled the logcount should not be 0.

**Comment by tobrun** [comment]
Upstream we just merged the ability to filter out categories from common SDK. I rather see us moving to that solution instead. We will however need to add the log configuration by default to disable log messages for users and remove the filter if we want to enable it again.

File: plugin-animation/src/main/java/com/mapbox/maps/plugin/animation/CameraAnimationsPluginImpl.kt (hunk 1 above)
Comment (kiryldz) [reply plugin-animation/src/main/java/com/mapbox/maps/plugin/animation/CameraAnimationsPluginImpl.kt:60]: I've set it to true in order to not make it a breaking change in terms of behaviour as now those logs are always enabled.

File: plugin-animation/src/test/java/com/mapbox/maps/plugin/animation/CameraAnimationsPluginImplTest.kt (hunk 2 above)
Comment (kiryldz) [reply plugin-animation/src/test/java/com/mapbox/maps/plugin/animation/CameraAnimationsPluginImplTest.kt:972]: Correct, that's why I'm using here `assertNotEquals` 😉

**Comment by tobrun** [comment]
> Upstream we just merged the ability to filter out categories from common SDK. I rather see us moving to that solution instead. We will however need to add the log configuration by default to disable log messages for users and remove the filter if we want to enable it again.

After internal convo, le'ts move forward on exposing debugMode. This can use the internal API under the hood.

File: plugin-animation/src/main/java/com/mapbox/maps/plugin/animation/CameraAnimationsPluginImpl.kt (hunk 1 above)
Comment (tobrun) [reply plugin-animation/src/main/java/com/mapbox/maps/plugin/animation/CameraAnimationsPluginImpl.kt:60]: I also would prefer false as default. I don't consider logs to be part of breaking changes.

Comment (kiryldz) [reply plugin-animation/src/main/java/com/mapbox/maps/plugin/animation/CameraAnimationsPluginImpl.kt:60]: thanks for your thoughts here, folks, will disable by default 👍

**Review by Chaoba — APPROVED** [review]
LGTM

**Review by tobrun — APPROVED** [review]

File: sdk-base/src/main/java/com/mapbox/maps/plugin/animation/CameraAnimationsPlugin.kt [hunk 3]
```diff
@@ -12,6 +12,12 @@ import com.mapbox.maps.plugin.MapPlugin
  */
 interface CameraAnimationsPlugin : MapPlugin {
 
+  /**
+   * If debug mode is enabled extra logs will be written about animation lifecycle and
+   * some other events that may be useful for debugging.
+   */
+  var debugMode: Boolean
```
Comment (pengdev) [inline sdk-base/src/main/java/com/mapbox/maps/plugin/animation/CameraAnimationsPlugin.kt:19]: shall we generate this config option for all the plugins? cc @kiryldz @tobrun @Chaoba

Comment (kiryldz) [reply sdk-base/src/main/java/com/mapbox/maps/plugin/animation/CameraAnimationsPlugin.kt:19]: No, I've taken a look at other plugins, there's no sense in adding it everywhere. We don't have that massive logs anywhere. That's why I closed internal tickets related to similar API for all the plugins and decided to implement it in camera only due to a lot of requests.

**Pull request merged**